ngraph/core/src/op/loop.cpp

```diff
@@ -43,6 +43,14 @@ bool op::v5::Loop::visit_attributes(AttributeVisitor& visitor)
     visitor.on_attribute("output_descriptions", m_output_descriptions);
     visitor.on_attribute("special_body_ports", m_special_body_ports);
 
+    for (const auto& output_description : m_output_descriptions)
```

Comment:
Why did we add this code only for serialization? I mean why didn't we have it before?

Reply:
@sdurawa Could you please explain in more detail? Why do we need this code?

Reply:
The reason we need `m_num_iterations` initialization is to produce static Concat output for Loop. When default value (-1) is used we will get one dynamic shape:
```
if (out_shape[axis].is_static() && m_num_iterations != -1)
{
    out_shape[axis] = Dimension{out_shape[axis].get_length() * m_num_iterations};
}
else
{
    out_shape[axis] = Dimension::dynamic();
}
```
It is required for serialization because XmlSerializer is checking outputs against dynamic shape:
```
for (auto o : node->outputs()) {
    NGRAPH_CHECK(o.get_partial_shape().is_static(),
        "Unsupported dynamic output shape in ", node);
```

Reply:
But isn't `validate_and_infer_type()` function of the loop called first? before this lines
```
 for (auto o : node->outputs()) {
    NGRAPH_CHECK(o.get_partial_shape().is_static(),
        "Unsupported dynamic output shape in ", node);

```
I expected that Concat output should be static and to m_num_iteration is set.

Reply:
No, first `visit_attributes()` is called, after that `validate_and_infer_types()`. And `m_num_iterations ` has to be set before validate is called, otherwise concat output will contain dynamic shape.

Reply:
Removing `m_num_iterations ` initialization as it calculated during `validate_and_infer_types()`.

Reply:
fixed